### PR TITLE
refactor: remove easyjson dependency

### PR DESCRIPTION
**Pull request by fabriziosestito**
## Description

Removes the easyjson dependency by bumping `policy-sdk` to the latest version.

## Test

Existing tests should pass.

## Additional Information

Artifact hub configuration was updated.

Related to: https://github.com/kubewarden/kubewarden-controller/issues/492

### CODE REVIEW

File: settings.go [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 	"github.com/kubewarden/gjson"
 	kubewarden "github.com/kubewarden/policy-sdk-go"
-	easyjson "github.com/mailru/easyjson"
 )
 
+type RawSettings struct {
+	DeniedLabels      []string          `json:"denied_labels"`
+	MandatoryLabels   []string          `json:"mandatory_labels"`
+	ConstrainedLabels map[string]string `json:"constrained_labels"`
+}
```
Comment (flavio) [inline settings.go:18]: We should merge `RawSettings` with `Settings`, this was just a workaround caused by easyjson not supporting properly `mapset`. Can you give a shot at unmarshalling the json straight into `Settings`?

**Review by viccuad — APPROVED** [review]
LGTM, module Flavio's comment.

File: settings.go (hunk 1 above)
Comment (fabriziosestito) [reply settings.go]: refactored in: [c322882](https://github.com/kubewarden/safe-labels-policy/commit/c322882e2d9be13787ade663d8bc50c50980055c).
Implemented `json.Unmarshaler` for `Settings`, but kept an anonymous struct similar to `RawSettings`to simplify the unmarshalling.

**Review by flavio — APPROVED** [review]
I love the new approach to handle `Settings` and `RawSettings` :clap:

**Comment by flavio** [comment]
I'm going to merge it, so that I can trigger a rebase of https://github.com/kubewarden/safe-labels-policy/pull/35 and merge it. Please wait for https://github.com/kubewarden/safe-labels-policy/pull/35 to be merged before tagging a new release

**Pull request merged**